src/PeriodicSelector.jl

```diff
@@ -33,7 +33,7 @@ struct PeriodicSelector <: DateSelector
 end
 
 
-function Iterators.partition(dates::StepRange{Date, Day}, s::PeriodicSelector)
+function Iterators.partition(dates::StepRange{Date, Day}, s::PeriodicSelector; bad_dates=[])
```

Comment:
I suggest renaming this to `exclude` and making default arg typed.

Here and everywhere else
```suggestion
function Iterators.partition(dates::StepRange{Date, Day}, s::PeriodicSelector; exclude=Date[])
```